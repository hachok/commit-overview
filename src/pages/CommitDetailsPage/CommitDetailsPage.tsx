import React, {Fragment, useEffect, useState} from 'react';
import {useHistory, useParams} from 'react-router-dom';
import {Box, Button, Card, CardContent, makeStyles, Typography} from '@material-ui/core';

import CommitFile from 'src/components/CommitFile/CommitFile';
import Commit from 'src/components/Commit/Commit';
import {API_URL} from 'src/constants';
import {CommitData} from 'src/types';

export interface FileInfo {
    sha: string
    filename: string
    patch: string
}

interface CommitInfo extends CommitData{
    files: FileInfo[]
}

interface Params {
    id: string
}

const useStyles = makeStyles(() => ({
    card: {
        width: '100%',
    }
}));

const CommitDetailsPage: React.FC = () => {
    const {id} = useParams<Params>();
    const [commit, setCommit] = useState<CommitInfo>();
    const history = useHistory();
    const classes = useStyles();

    useEffect(() => {
        const fetchCommitDetails = async () => {
            const res = await fetch(`${API_URL}/commits/` + id);
            return await res.json();
        };

        fetchCommitDetails().then((commit: CommitInfo) => setCommit(commit));
    }, [id]);

    return (
        <Fragment>
            <Button variant="contained" onClick={() => history.push('/')}>Go back</Button>
            {commit && <Fragment>
                <Box m={2} width="100%">
                    <Typography variant="h6" color="inherit" noWrap>Info: </Typography>
                </Box>
                <Card className={classes.card}>
                    <CardContent>
                        <Commit info={commit}/>
                        <Box mt={2}>
                                <pre>
                                    <code>{commit.commit.message}</code>
                                </pre>
                        </Box>
                    </CardContent>
                </Card>
                <Box m={2}>
                    <Typography variant="h6" color="inherit" noWrap>Files: </Typography>
                </Box>
                {!!commit.files.length && commit.files.map((file: FileInfo) => <CommitFile key={file.sha}
                                                                                           info={file}/>)}
            </Fragment>}
        </Fragment>
    );
};

export default CommitDetailsPage;
