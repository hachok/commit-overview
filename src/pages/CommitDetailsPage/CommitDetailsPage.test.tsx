import {cleanup, render} from '@testing-library/react';
import CommitDetailsPage from 'src/pages/CommitDetailsPage/CommitDetailsPage';
import '@testing-library/jest-dom/extend-expect'
import fetch from 'jest-fetch-mock';

jest.mock('react-router-dom', () => ({
    useParams: () => ({
        id: 'd74559746c03d9f620e4fd338c70ce319070e0be',
    }),
    useHistory: () => ({
        push: jest.fn(),
    }),
}));

const mockData = {
    'sha': 'd74559746c03d9f620e4fd338c70ce319070e0be',
    'commit': {
        'author': {
            'name': 'Ricky',
            'email': 'rickhanlonii@gmail.com',
            'date': '2021-03-07T23:43:29Z'
        },
        'message': 'Use update lane priority to set pending updates on roots (#20918)',
    },
    'author': {
        'avatar_url': 'https://avatars.githubusercontent.com/u/2440089?v=4',
    },
    'files': [
        {
            'sha': '77bd914eaa5fc90a9f808e1a5a8acf14ed96e78c',
            'filename': 'packages/react-reconciler/src/ReactFiberWorkLoop.old.js',
            'patch': '@@ -598,6 +598,27 @@ export function scheduleUpdateOnFiber(\n         flushSyncCallbackQueue();\n       }\n     }\n+  } else if (decoupleUpdatePriorityFromScheduler) {\n+    const updateLanePriority = getCurrentUpdateLanePriority();\n+\n+    // Schedule a discrete update but only if it\'s not Sync.\n+    if (\n+      (executionContext & DiscreteEventContext) !== NoContext &&\n+      // Only updates at user-blocking priority or greater are considered\n+      // discrete, even inside a discrete event.\n+      updateLanePriority === InputDiscreteLanePriority\n+    ) {\n+      // This is the result of a discrete event. Track the lowest priority\n+      // discrete update per root so we can flush them early, if needed.\n+      if (rootsWithPendingDiscreteUpdates === null) {\n+        rootsWithPendingDiscreteUpdates = new Set([root]);\n+      } else {\n+        rootsWithPendingDiscreteUpdates.add(root);\n+      }\n+    }\n+    // Schedule other updates after in case the callback is sync.\n+    ensureRootIsScheduled(root, eventTime);\n+    schedulePendingInteractions(root, lane);\n   } else {\n     // Schedule a discrete update but only if it\'s not Sync.\n     if ('
        }
    ]
};

beforeEach(() => {
    fetch.resetMocks();
});

afterEach(cleanup);

test('it should render component correctly', async () => {
    fetch.mockResponseOnce(JSON.stringify(mockData));
    const rendered = render(<CommitDetailsPage/>)

    await rendered.findByText('Ricky');

    expect(rendered).toMatchSnapshot();
})
