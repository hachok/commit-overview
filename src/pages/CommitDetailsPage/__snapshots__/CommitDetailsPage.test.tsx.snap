// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Go back
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiBox-root MuiBox-root-2"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
        >
          Info: 
        </h6>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div
            class="makeStyles-wrapper-3"
            data-testid="commit-element"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
            >
              <div
                class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circle"
                >
                  <img
                    alt="Avatar"
                    class="MuiAvatar-img"
                    src="https://avatars.githubusercontent.com/u/2440089?v=4"
                  />
                </div>
              </div>
              <div
                class="MuiListItemText-root MuiListItemText-multiline"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  d74559746c03d9f620e4fd338c70ce319070e0be
                </span>
                <p
                  class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                >
                  <span
                    class="MuiTypography-root makeStyles-inline-4 MuiTypography-body2 MuiTypography-colorTextPrimary"
                  >
                    Ricky
                  </span>
                   - 
                  08-03-2021
                </p>
              </div>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-inset"
            />
          </div>
          <div
            class="MuiBox-root MuiBox-root-5"
          >
            <pre>
              <code>
                Use update lane priority to set pending updates on roots (#20918)
              </code>
            </pre>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-6"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
        >
          Files: 
        </h6>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-7 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <p
            class="MuiTypography-root makeStyles-title-8 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            packages/react-reconciler/src/ReactFiberWorkLoop.old.js
          </p>
          <pre>
            <code>
              @@ -598,6 +598,27 @@ export function scheduleUpdateOnFiber(
         flushSyncCallbackQueue();
       }
     }
+  } else if (decoupleUpdatePriorityFromScheduler) {
+    const updateLanePriority = getCurrentUpdateLanePriority();
+
+    // Schedule a discrete update but only if it's not Sync.
+    if (
+      (executionContext & DiscreteEventContext) !== NoContext &&
+      // Only updates at user-blocking priority or greater are considered
+      // discrete, even inside a discrete event.
+      updateLanePriority === InputDiscreteLanePriority
+    ) {
+      // This is the result of a discrete event. Track the lowest priority
+      // discrete update per root so we can flush them early, if needed.
+      if (rootsWithPendingDiscreteUpdates === null) {
+        rootsWithPendingDiscreteUpdates = new Set([root]);
+      } else {
+        rootsWithPendingDiscreteUpdates.add(root);
+      }
+    }
+    // Schedule other updates after in case the callback is sync.
+    ensureRootIsScheduled(root, eventTime);
+    schedulePendingInteractions(root, lane);
   } else {
     // Schedule a discrete update but only if it's not Sync.
     if (
            </code>
          </pre>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Go back
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
      >
        Info: 
      </h6>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <div
          class="makeStyles-wrapper-3"
          data-testid="commit-element"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
          >
            <div
              class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle"
              >
                <img
                  alt="Avatar"
                  class="MuiAvatar-img"
                  src="https://avatars.githubusercontent.com/u/2440089?v=4"
                />
              </div>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                d74559746c03d9f620e4fd338c70ce319070e0be
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                <span
                  class="MuiTypography-root makeStyles-inline-4 MuiTypography-body2 MuiTypography-colorTextPrimary"
                >
                  Ricky
                </span>
                 - 
                08-03-2021
              </p>
            </div>
          </li>
          <hr
            class="MuiDivider-root MuiDivider-inset"
          />
        </div>
        <div
          class="MuiBox-root MuiBox-root-5"
        >
          <pre>
            <code>
              Use update lane priority to set pending updates on roots (#20918)
            </code>
          </pre>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-6"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
      >
        Files: 
      </h6>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-7 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root makeStyles-title-8 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
        >
          packages/react-reconciler/src/ReactFiberWorkLoop.old.js
        </p>
        <pre>
          <code>
            @@ -598,6 +598,27 @@ export function scheduleUpdateOnFiber(
         flushSyncCallbackQueue();
       }
     }
+  } else if (decoupleUpdatePriorityFromScheduler) {
+    const updateLanePriority = getCurrentUpdateLanePriority();
+
+    // Schedule a discrete update but only if it's not Sync.
+    if (
+      (executionContext & DiscreteEventContext) !== NoContext &&
+      // Only updates at user-blocking priority or greater are considered
+      // discrete, even inside a discrete event.
+      updateLanePriority === InputDiscreteLanePriority
+    ) {
+      // This is the result of a discrete event. Track the lowest priority
+      // discrete update per root so we can flush them early, if needed.
+      if (rootsWithPendingDiscreteUpdates === null) {
+        rootsWithPendingDiscreteUpdates = new Set([root]);
+      } else {
+        rootsWithPendingDiscreteUpdates.add(root);
+      }
+    }
+    // Schedule other updates after in case the callback is sync.
+    ensureRootIsScheduled(root, eventTime);
+    schedulePendingInteractions(root, lane);
   } else {
     // Schedule a discrete update but only if it's not Sync.
     if (
          </code>
        </pre>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
